src/index.js

```diff
@@ -109,7 +109,10 @@ async function getCurrentOrders() {
 
         // TKD seems to correspond to ebillet but maybe there are other types of delivery modes
         // which allow to download a file
-        if ( body.order.trainFolders[code].deliveryMode.type === 'TKD') {
+        if (
+          body.order.trainFolders[code].deliveryMode.type === 'TKD' &&
+          body.order.trainFolders[code].ticketlessStatus !== 'FULL' // in this case the is no PDF ebillet to print (billet on smartphone)
```

Comment:
There is a typo in the comment line `s/the/there/`

And a frenchism `s/billet/ticket/`

Reply:
To avoid _hard-to-maintain-synchronised_ comments, you could place the obscur condition into a _well-named_ function:

```js
function isThereAPhysicalTicket (ticketInformation) {
    return ticketInformation.deliveryMode.type === "TKD"
        && ticketInformation.ticketlessStatus !== "FULL";
}
```
and in the `if` block:
```js
if(isThereAPhysicalTicket(body.order.trainFolders[code])) {
    // ...
}
```

And you can be as much precise you need:

```js
function isTheTicketOnlyOnSmartphone (status) {
    return status === "FULL";
}


function isThereAPhysicalTicket (ticketInformation) {
    return ticketInformation.deliveryMode.type === "TKD"
        && !isTheTicketOnlyOnSmartphone(ticketInformation.ticketlessStatus);
}
```

Reply:
Thanks for this, I like it better now @enguerran